File: 66.加一.cs
/*
 * @lc app=leetcode.cn id=66 lang=csharp
 *
 * [66] 加一
 */

// @lc code=start
public partial class Solution
{
    public int[] PlusOne(int[] digits)
    {
        int length = digits.Length;
        bool add = true;
        for (int i = length - 1; i >= 0; i--)
        {
            int num = digits[i];
            if (add)
            {
                num += 1;
                if (num == 10)
                {
                    add = true;
                    num = 0;
                }
                else
                {
                    add = false;
                }
                digits[i] = num;
            }
            else
            {
                break;
            }
        }
        if (!add)
        {
            return digits;
        }
        int[] result = new int[length + 1];
        result[0] = 1;
        Array.Copy(digits, 0, result, 1, length);
        return result;
    }
}
// @lc code=end

File: 66.加一.rs
/*
 * @lc app=leetcode.cn id=66 lang=rust
 *
 * [66] 加一
 */

// @lc code=start
impl Solution {
    pub fn plus_one(digits: Vec<i32>) -> Vec<i32> {
        let add = true;
        let len = digits.len();
        let mut index = len - 1;
        while (add && index >= 0) {
            let num = digits[index];
            num = num + 1;
            if (num == 10) {
                add = true;
                num = 0;
            } else {
                add = false;
            }
            digits[index] = num;
            index -= 1;
        }
        if (!add) {
            return digits;
        }
        let digits = vec![];
        return digits;
    }
}
// @lc code=end
